presto-hive/src/main/java/com/facebook/presto/hive/HivePartialAggregationPushdown.java

```diff
@@ -188,29 +192,29 @@ private boolean isAggregationPushdownSupported(AggregationNode partialAggregatio
             return true;
         }
 
-        private boolean isSupportedOrcTypeForMinMax(Type type)
+        private boolean isNotSupportedOrcTypeForMinMax(Type type)
         {
-            return !BOOLEAN.equals(type) &&
-                    type.getJavaType() != boolean.class &&
-                    !isRowType(type) &&
-                    !isArrayType(type) &&
-                    !isMapType(type) &&
-                    !TINYINT.equals(type) &&
-                    !VARBINARY.equals(type) &&
-                    !TIMESTAMP.equals(type) &&
-                    ((!VARBINARY.equals(type) && !VARCHAR.equals(type)) ||
-                            isPartialAggregationPushdownForVariableLengthDatatypesEnabled(session));
+            return BOOLEAN.equals(type) ||
+                    type.getJavaType() == boolean.class ||
+                    isRowType(type) ||
+                    isArrayType(type) ||
+                    isMapType(type) ||
+                    TINYINT.equals(type) ||
+                    VARBINARY.equals(type) ||
+                    TIMESTAMP.equals(type) ||
+                    ((VARBINARY.equals(type) || VARCHAR.equals(type)) &&
```

Comment:
Adding an extra pair of (...) around varbinary && isPart... would increase readability a bit and will reduce a risk of accidentally breaking this sensitive pair coming after a bunch of ORs.

Reply:
Added a new method `isNotSupportedForVariableLengthDataType` which makes this part more clear.